server/rest/src/main/java/org/infinispan/rest/server/operations/mediatypes/printers/JSONOutputPrinter.java

```diff
@@ -0,0 +1,50 @@
+package org.infinispan.rest.server.operations.mediatypes.printers;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.stream.Collectors;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.infinispan.CacheSet;
+import org.infinispan.rest.logging.Log;
+import org.infinispan.rest.server.operations.exceptions.ServerInternalException;
+import org.infinispan.rest.server.operations.mediatypes.OutputPrinter;
+import org.infinispan.util.logging.LogFactory;
+
+public class JSONOutputPrinter implements OutputPrinter {
+
+   protected final static Log logger = LogFactory.getLog(JSONOutputPrinter.class, Log.class);
+
+   private static class JsonMapperHolder {
+      public static final ObjectMapper jsonMapper = new ObjectMapper();
+   }
+
+   @Override
+   public byte[] print(String cacheName, CacheSet<?> keys) {
+      return keys.stream()
+            .map(b -> Escaper.escapeJson(b.toString()))
+            .collect(Collectors.joining("keys=", "", "]"))
```

Comment:
The only think you have to watch out for is if this is a backed by a distributed cache. In which case the collector here isn't serializable. You can use `CacheCollectors` to be safe unless you know this won't go remote.